packages/nx/src/config/workspaces.ts

```diff
@@ -637,6 +640,9 @@ export function buildWorkspaceConfigurationFromGlobs(
       // directory as a package.json should overwrite the inferred package.json
       // project configuration.
       const configuration = readJson(file);
+
+      configuration.root = directory;
```

Comment:
Given that we are now doing a little bit more, maybe we should abstract this out. Something like below. This isn't super important right now, but if we add any token support in the file, it will get more so.

```typescript
function buildProjectConfigurationFromProjectJson(filePath) {
	const c = readJson<ProjectConfiguration>(filePath)
	c.root = dirname(filePath)
	return c
}
```